vms/platformvm/state/camino_test.go

```diff
@@ -162,6 +268,98 @@ func TestSyncGenesis(t *testing.T) {
 	}
 }
 
+func defaultConfig(lockModeBondDeposit bool, nodeID ids.NodeID, deposit bool) root_genesis.Config {
+	depositOffers := []genesis.DepositOffer{
+		{
+			OfferID:                 id,
+			InterestRateNominator:   1,
+			Start:                   2,
+			End:                     3,
+			MinAmount:               4,
+			MinDuration:             5,
+			MaxDuration:             6,
+			UnlockPeriodDuration:    7,
+			NoRewardsPeriodDuration: 2,
+			Flags:                   9,
+		},
+	}
+
+	if deposit {
+		id0, _ = depositOffers[0].ID()
```

Comment:
id != depositOffers[0].ID(), there are different values. If you set DepositOfferID with the value of id eventually will be saved in a map where the key of that map is the id value. There are several places in the code though, that tries to fetch the value in the map by indexing the map with offer.ID() BUT offer.ID() == depositOffers[0].ID() where it does not have the same value as id.

In other words, if we use id's value as DepositOfferID it will store the offer in a map with the wrong key and eventually will through "non existing deposit offer" error when we try to retrieve the data from the map.

Example: platformvm/api/camino.go>makeUTXOAndDeposit line 367. The code tries to retrieve map's data by using the value of "id" as key, while the correct key is offer.ID() as initiated in buildCaminoGenesis function line 79.

Reply:
if genesis is valid, that offer.id == offer.ID(), so, plz, use valid offer id here - that will solve the issue, if I got it right